lib/ClangImporter/ImportDecl.cpp

```diff
@@ -4511,15 +4511,15 @@ namespace {
     }
 
     Decl *VisitObjCInterfaceDecl(const clang::ObjCInterfaceDecl *decl) {
-      auto createRootClass = [=](Identifier name,
-                                 DeclContext *dc = nullptr) -> ClassDecl * {
+      auto createFakeRootClass = [=](Identifier name,
+                                     DeclContext *dc = nullptr) -> ClassDecl * {
         if (!dc) {
           dc = Impl.getClangModuleForDecl(decl->getCanonicalDecl(),
                                           /*allowForwardDeclaration=*/true);
         }
 
         auto result = Impl.createDeclWithClangNode<ClassDecl>(decl,
-                                                        AccessLevel::Open,
+                                                        AccessLevel::Public,
```

Comment:
This part isn't obvious.

Reply:
Ah, this lambda is only used for forward declarations and the "Protocol" type. You could never meaningfully subclass those anyway, and most of the time you'd run into trouble anyway because they have no initializers.

Reply:
Okay, that makes sense to me.